app/src/main/java/com/github/wnder/picture/ExistingPicture.java

```diff
@@ -31,7 +34,11 @@
 
     public ExistingPicture(String uniqueId){
         super(uniqueId);
-        addToUserGuessedPictures(GlobalUser.getUser().getName());
+        User user = GlobalUser.getUser();
+        if(user instanceof SignedInUser){
+
+            addToUserGuessedPictures(user.getName());
+        }
```

Comment:
Do we want to add to the guessed pictures as soon as a user "sees" a picture ? Or do we want to wait for him to press on the "guess picture" button ? (or maybe I didn't understand something)

Reply:
Good question! Then we also need to decide what to do with skipped and reported pictures

Reply:
Skipped, reported or guessed pictures are all pictures that the user has already encounter, so they should be the one added to "guessedPictures".

Reply:
Then I guess @BoostedBoat can approve the review?

Reply:
I agree with @leonardvaney and @JeremyDezalos after discussing with him agrees too.